[Ivan Ho] iP

ivanhlb opened the pull request.

Commit:
Add Gradle support

ivanhlb committed:
Basic Update with reference to https://github.com/ivanhlb/duke

ivanhlb committed:
Minor editing to increase readability.

l-shihao commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,97 @@
+ * Primarily for printing messages to console, also passes input to Parser.
+ */
+public class Ui {
+    private Scanner input;
```
great, learn from you, always kinda unsure about how to encapsulate readin 👏

l-shihao commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,97 @@
+        String inputStr = "";
+        while (inputStr.isEmpty() || inputStr.isEmpty()) {
+            try {
+                inputStr = input.nextLine().strip();
```
I'm using `trim()`, after research, `strip()` is better

l-shihao commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,37 @@
+
+import exception.DukeMissingDescException;
+
+public abstract class Task {
```
How do you think a static property to store the `totalNumOfTask`?
not sure if java properties can be computed properties, so that I hope to give each task a unique but dynamically changing `taskNo` property 🤔 
now my `taskNo` is initiated by `= totalNumOfTask`, but it works only because there is no delete task function in Duke 

also thinking about if the `totalNumOfTask` can be more dynamically mutated to be `totalNumOfUndoneTask` which is more useful

l-shihao commented on src/main/java/exception/DukeUnknownException.java:
```diff
@@ -0,0 +1,8 @@
+public class DukeUnknownException extends DukeException {
+    @Override
+    public String getMessage() {
+        return "OOPS!!! I'm sorry, but I don't know what that means :-(";
```
not sure whether too overly done if instead of the String here, replace it with something like: 
`Ui.dukeUnknownExceptionMessage()`? maybe a static method for simplicity

l-shihao commented on src/main/java/command/UpdateCommand.java:
```diff
@@ -0,0 +1,89 @@
+import task.*;
+import ui.Ui;
+
+public class UpdateCommand extends Command {
```
first glance I thought this is a command for Add new commands, so that there will be a data structure that holds all the commands, and others programmer can create new sub class command and add to command map elsewhere. 🤔

l-shihao commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,45 @@
+                String fullCommand = ui.readCommand();
+                Command c = Parser.parseInput(fullCommand);
+                c.execute(tasks, ui, storage);
+                running = !c.isExit();
```
this is why you put `isExit()` in abstract Command class, kinda nice! 
I didn't use loop here, makes my life harder to use function to exit the recursive call 😮‍💨

l-shihao commented:
Comment on coding standard violations only 😱 I'm screwed.

zcnmashleu95 reviewed (commented):
Generally, the readability is great. Methods are named with verbs, exceptions are also thrown.
Readability and Naming are fine

zcnmashleu95 commented on src/main/java/command/UpdateCommand.java:
```diff
@@ -0,0 +1,89 @@
+    public Operation getOperation() {
+        return operation;
+    }
+
```
Use of Enum is great. Naming is also intuitive

ivanhlb committed:
Level already implemented as of TIC2002.
fixed string description for MissingParamException that was newly added.

ivanhlb committed:
Merge branch 'branch-Level-5'

ivanhlb committed:
Task already marked abstract.
WIP for Deadline and CheckCommand to be more flexible with dates (accept string values)

l-shihao commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,37 @@
+public abstract class Task {
+    protected boolean isDone;
+    protected String description;
+    public static final char CHECKMARK = '\u2713', CROSSMARK = '\u274C';
```
I like the all caps for constant, good 😄

ivanhlb committed:
Merge pull request #1 from ivanhlb/branch-A-AbstractClasses

Task already marked abstract.

ivanhlb committed:
For Level-4:
Fixed Event and Deadline so that it can take plain strings (which are not dateTime format).

ivanhlb committed:
Fixes:
- Changed comment in TaskList to standard string as javac was complaining about it.

Changes:
- Readded ECHO OFF for textTesting. (Previously Removed for checking output for testing)
- WIP for A-TextUiTesting
- Added some fixes for Unicode support for A-TextUiTesting

ivanhlb committed:
Fixed Storage.java for Level 7 (saving to .txt file)

ivanhlb committed:
Added some JavaDocs.

ivanhlb committed:
Merge branch 'branch-A-JavaDoc'

ivanhlb committed:
Merge branch 'branch-Level-7'

ivanhlb committed:
GFMD

ivanhlb committed:
shifted all the separate packages under duke.*.

ivanhlb committed:
Delete already inplemented. Added Javadoc to class to register.

ivanhlb committed:
Merge pull request #2 from ivanhlb/branch-Level-6

Level 6

ivanhlb committed:
More OOP. Already implemented in TIC2002, added Javadoc to earmark key files for it.

ivanhlb committed:
Implemented some JUnit test for Task.

ivanhlb committed:
Level 9 already implemented in TIC2002

ivanhlb committed:
Merge pull request #3 from ivanhlb/branch-A-JUnit

branch-A-JUnit

ivanhlb committed:
Merge pull request #4 from ivanhlb/branch-Level-9

Level 9

ivanhlb committed:
Added Javadoc to Deadline and Event, Level-8 already implemented.

ivanhlb committed:
Merge pull request #5 from ivanhlb/branch-Level-8

Added Javadoc to Deadline and Event, Level-8 already implemented.

ivanhlb committed:
Set theme jekyll-theme-minimal

ivanhlb committed:
Added preliminary user guide.

ivanhlb committed:
Imported classes should always be listed explicitly.

ivanhlb committed:
Begin TIC4002

ivanhlb committed:
Merge branch 'add-gradle-support'

ivanhlb committed:
Added Gradle support. Able to run.

ivanhlb committed:
Added Gradle runtest config.

ivanhlb committed:
WIP to conform to checkstyle

ivanhlb committed:
Compliance to Checkstyle rules.

ivanhlb committed:
WIP for GUI version

ivanhlb committed:
WIP, left editCommand

ivanhlb committed:
Fix for editing commands. Fully Working GUI.

ivanhlb committed:
Merge branch 'Level-10'

ivanhlb committed:
updated build.gradle and files for gradle jar build.

ivanhlb committed:
Create gradle.yml

ivanhlb committed:
Fix for checkStyle, removed unused import.

ivanhlb committed:
Merge pull request #7 from ivanhlb/A-CI

Create gradle.yml

ivanhlb committed:
Implemented sort command, also fixed finicky save/load function by replacing nonstandard checkmark/crossmark with o/x.

ivanhlb committed:
Refactored inline EventHandler to its own class.

ivanhlb committed:
Updated InputEventHandler to be more readable.

ivanhlb committed:
WIP.
